core/src/main/scala/akka/kafka/internal/MessageBuilder.scala

```diff
@@ -187,8 +187,6 @@ private[kafka] final class CommittableOffsetBatchImpl(
     val metadata = newOffset match {
       case offset: CommittableOffsetMetadata =>
         offset.metadata
-      case _ =>
```

Comment:
`CommittableOffsetMetadata` is the only subtype of a `sealed trait` - so it should be ok to be removed as scala3 complained about it.